src/ZkFold/Base/Algebra/Basic/Class.hs

```diff
@@ -162,7 +162,7 @@ class MultiplicativeMonoid b => Scale b a where
 
 instance MultiplicativeMonoid a => Scale a a
 
-instance {-# OVERLAPPABLE #-} (Scale b a, Functor f) => Scale b (f a) where
+instance {-# OVERLAPPABLE #-} (Scale b a, Functor f, MultiplicativeMonoid b, fa ~ f a) => Scale b fa where
```

Comment:
This instance was annoying me as well, I was going to submit a PR which removed this instance in addition to some other improvements to numeric hierarchy